plugins/org.eclipse.n4js/src/org/eclipse/n4js/resource/N4JSResourceDescriptionStrategy.java

```diff
@@ -47,48 +47,55 @@
 	 * Main module has different shadowing rules in the scoping. Note: Only {@link IEObjectDescription object
 	 * descriptions} representing a {@link TModule} have this user data.
 	 */
-	public static final String MAIN_MODULE_KEY = "MAIN_MODULE_KEY";
+	public static final String MAIN_MODULE_KEY = "MAIN_MODULE";
+	private static final boolean MAIN_MODULE_DEFAULT = false;
 
 	/**
 	 * The user data key for the type access modifier - used to compute visibility without creating a resource just from
 	 * the proxy in the index.
 	 */
-	public static final String ACCESS_MODIFIERY_KEY = "ACCESS_MODIFIER_KEY";
+	private static final String TYPE_ACCESS_MODIFIER_KEY = "TYPE_ACCESS_MODIFIER";
+	private static final TypeAccessModifier TYPE_ACCESS_MODIFIER_DEFAULT = TypeAccessModifier.PUBLIC;
 
 	/**
 	 * User data to store the {@link TClass#isFinal() final} property of a {@link TClass} in the index without resolving
 	 * the actual proxy. Used by N4JS type search. Note: Only {@link IEObjectDescription object description}s
 	 * representing a {@link TClass} has this user data.
 	 */
-	public static final String FINAL_KEY = "FINAL_KEY";
+	public static final String FINAL_KEY = "FINAL";
```

Comment:
nice clean up, reducing the accessibility of these constants!!
But why keep `public` for `FINAL_KEY` and `MAIN_MODULE_KEY`?

Reply:
Thanks, right, I change `FINAL_KEY` to private. Regarding the `MAIN_MODULE_KEY` key: it is used from `UserdataMapper` and to refactor this would take a deeper look into the code.